zone.go

```diff
@@ -738,3 +740,18 @@ func (api *API) UpdateFallbackOrigin(zoneID string, fbo FallbackOrigin) (*Fallba
 
 	return response, nil
 }
+
+// normalizeZoneName tries to convert IDNs (international domain names)
+// from Punycode to Unicode form. If the given zone name is not represented
+// as Punycode, or converting fails (for invalid representations), it
+// is returned unchanged.
+//
+// Note: conversion errors are silently discarded.
+func normalizeZoneName(name string) string {
+	if strings.Contains(name, "xn--") {
```

Comment:
I don't think we need to check if the string contains the "xn--" characters. Instead we can run all values through `idna.ToUnicode`.

Reply:
Yeah, you're right. I thougth it might avoid some unnecessary processing overhead, but I should have benchmarked it first.

Here are my current results ([code](https://gist.github.com/dmke/d0ae099ab00873ca5281350bed91a6b3)):

```
ToUnicode(punycode) w/ prefix check       3000000	       472 ns/op
ToUnicode(punycode) w/o prefix check      3000000	       472 ns/op
ToUnicode(unicode) w/ prefix check       200000000	         9.99 ns/op
ToUnicode(unicode) w/o prefix check      200000000	        10.00 ns/op
```

The prefix check saves (at most) a few dozen picoseconds :-)